### ISSUE TITLE
Why don't we use UnauthorizedError

### ISSUE THREAD

**Issue by gene1wood** [mozilla-iam/auth0-deploy#211]
In all the Auth0 docs and code I can find, the way a user login is denied is by returning a callback, passing to it a new `UnauthorizedError` instance.

https://auth0.com/docs/rules/current#deny-access-based-on-a-condition

In our code we don't do that. Why is that and how are we conveying to Auth0 the user should be denied.

**Comment by gdestuynder** [comment]
not all RPs support this - in fact, nearly none do
this causes the error msg to be shown sometimes in the URL bar, sometimes not at all

we use it sometimes when things would "completely fail anyway", but not where we want to convey something to the user, and they would most likely never see it

**Comment by gene1wood** [comment]
Got it, makes sense.

Ok, I did some more digging and here's what I found

When I compare cases where we grant access or deny access I see one difference
* To deny access we return [`callback(null, user, global.postError('acodehere', context)`](https://github.com/mozilla-iam/auth0-deploy/blob/2d3d7fb6e4b24586f3810aeaf7ad7ebb962fb9df/rules/AccessRules.js#L166)
* To grant access we return [`callback(null, user, context)`](https://github.com/mozilla-iam/auth0-deploy/blob/2d3d7fb6e4b24586f3810aeaf7ad7ebb962fb9df/rules/AccessRules.js#L225)

I Google for `postError` with no luck then realized that it's defined [earlier in our rule set in Global-Function-Declarations.js](https://github.com/mozilla-iam/auth0-deploy/blob/5a4fb1341a6504b5017f071a1a9101195f03bd0e/rules/Global-Function-Declarations.js#L11-L32)

`postError` takes the [`context`](https://auth0.com/docs/rules/current/context) variable and override the `redirect` attribute so the user gets sent to the SSO dashboard with our JWT signed error message instead of the RP. The reason this achieves denial of the authentication attempt is that the `code` that would have been issued to the user to pass to the RP is thrown away/lost when the user is sent to SSO dashboard.
